### PR TITLE
Changed ZooKeeper path to use table UUID

### PR DESCRIPTION
**Pull request by estaine**

### CODE REVIEW

**Comment by darkfrog26** [comment]
@sazonov any reservations with merging this into your branch?

**Comment by estaine** [comment]
Hey, this PR moves Zookeeper path to a parameter. We use a {uuid} macros-based Zookeeper paths in order not to make an extra dependency between a table name and its path in Zookeper.
The default behaviour (when no config parameter is set) is left intact.

File: flyway-community-db-support/flyway-database-clickhouse/src/main/java/org/flywaydb/community/database/clickhouse/ClickHouseConfigurationExtension.java [hunk 1]
```diff
@@ -23,8 +23,10 @@
 @Getter
 public class ClickHouseConfigurationExtension implements ConfigurationExtension {
     private static final String CLUSTER_NAME = "flyway.clickhouse.clusterName";
+    private static final String ZOOKEEPER_PATH = "flyway.zookeeper.path";
```
Comment (sazonov) [inline flyway-community-db-support/flyway-database-clickhouse/src/main/java/org/flywaydb/community/database/clickhouse/ClickHouseConfigurationExtension.java]: Let's change the constant to "flyway.clickhouse.zookeeperPath" besause it applies to ClickHouse only.

File: flyway-community-db-support/flyway-database-clickhouse/src/main/java/org/flywaydb/community/database/clickhouse/ClickHouseConfigurationExtension.java [hunk 2]
```diff
@@ -37,13 +39,21 @@ public void extractParametersFromConfiguration(Map<String, String> configuration
...
+            this.zookeeperPath = zookeeperPath;
+        }
     }
 
     @Override
     public String getConfigurationParameterFromEnvironmentVariable(String environmentVariable) {
         if ("FLYWAY_CLICKHOUSE_CLUSTER_NAME".equals(environmentVariable)) {
             return CLUSTER_NAME;
         }
+        if ("FLYWAY_ZOOKEEPER_PATH".equals(environmentVariable)) {
```
Comment (sazonov) [inline flyway-community-db-support/flyway-database-clickhouse/src/main/java/org/flywaydb/community/database/clickhouse/ClickHouseConfigurationExtension.java]: FLYWAY_ZOOKEEPER_PATH -> FLYWAY_CLICKHOUSE_ZOOKEEPER_PATH

File: flyway-community-db-support/flyway-database-clickhouse/src/main/java/org/flywaydb/community/database/clickhouse/ClickHouseDatabase.java [hunk 3]
```diff
@@ -93,7 +100,11 @@ public String getRawCreateScript(Table table, boolean baseline) {
         String engine;
 
         if (isClustered) {
-            engine = "ReplicatedMergeTree('/clickhouse/tables/{shard}/{database}/{table}', '{replica}')";
+            if (hasCustomZookeeperPath) {
+                engine = "ReplicatedMergeTree('" + zookeeperPath + "', '{replica}')";
```
Comment (sazonov) [inline flyway-community-db-support/flyway-database-clickhouse/src/main/java/org/flywaydb/community/database/clickhouse/ClickHouseDatabase.java]: '/clickhouse/tables/{shard}/{database}/{table}' this should be the default value for the new configuration param. I would suggest to set this value to ClickHouseConfigurationExtension::zookeeperPath as initial value so we keep current line for all cases and remove all redundant conditions.

**Review by sazonov — CHANGES_REQUESTED** [review]

**Comment by sazonov** [comment]
@estaine thank you for the work.

**Comment by estaine** [comment]
@sazonov Thank you so much for such a quick answer.
All your comments make sense and I've just updated my pull request

**Pull request merged**